app/models/miq_server/worker_management/heartbeat.rb

```diff
@@ -75,13 +81,23 @@ def post_message_for_workers(class_name = nil, resync_needed = false, sync_messa
 
   # Get the latest heartbeat between the SQL and memory (updated via DRb)
   def validate_heartbeat(w)
+    last_heartbeat = workers_last_heartbeat(w.pid)
+
     if w.last_heartbeat.nil?
-      w.last_heartbeat ||= @workers[w.pid][:last_heartbeat] if @workers.key?(w.pid)
-      w.last_heartbeat ||= Time.now.utc
```

Comment:
Note, we weren't saving `w` previously and the caller, `validate_worker`, [here](https://github.com/ManageIQ/manageiq/blob/3128981e03482b71f7a9f74bff0dbd6f4c942b67/app/models/miq_server/worker_management/monitor/validation.rb#L11) wasn't saving the record.  To be consistent with legs of the conditional, I changed it to now save the record.   Without the save, the tests would fail!